File: backend/providers/nous.js
module.exports = {

  name: 'nous',
  aliases: ['nous-research', 'nous-portal'],
  displayName: 'Nous Research (Portal)',
  description: 'Nous Portal API — hermes-4 and hermes-3 models',
  baseUrl: 'https://api.nousresearch.com/v1',
  apiMode: 'openai_chat',
  envVars: ['NOUS_API_KEY', 'NOUS_BASE_URL'],
  authType: 'api_key',
  defaultModel: 'hermes-4',
  fallbackModels: ['hermes-4-flash', 'hermes-3', 'hermes-3-flash'],
  defaultMaxTokens: 8192,
  signupUrl: 'https://nousresearch.com',
  supportsHealthCheck: true,
  modelProfiles: {
    'hermes-4': { supportsReasoning: true, supportsThinking: false, combinedBudget: false, contextWindow: 131072, maxOutputTokens: 8192 },
    'hermes-4-flash': { supportsReasoning: false, supportsThinking: false, combinedBudget: false, contextWindow: 131072, maxOutputTokens: 8192 },
    'hermes-3': { supportsReasoning: false, supportsThinking: false, combinedBudget: false, contextWindow: 131072, maxOutputTokens: 8192 },
    'hermes-3-flash': { supportsReasoning: false, supportsThinking: false, combinedBudget: false, contextWindow: 131072, maxOutputTokens: 8192 },
    '*': { supportsReasoning: false, supportsThinking: false, combinedBudget: false, contextWindow: 131072, maxOutputTokens: 4096 },
  },

};

File: backend-py/app/providers/nous.py
"""
Provider config for Nous Research (Portal).
"""
from __future__ import annotations
INFO: dict[str, object] = {'name': 'Nous Research (Portal)', 'aliases': ['nous-research', 'nous-portal'], 'display_name': 'Nous Research (Portal)', 'description': 'Nous Portal API — hermes-4 and hermes-3 models', 'base_url': 'https://api.nousresearch.com/v1', 'api_mode': 'openai_chat', 'env_vars': ['NOUS_API_KEY', 'NOUS_BASE_URL'], 'auth_type': 'api_key', 'default_model': 'hermes-4', 'fallback_models': ['hermes-4-flash', 'hermes-3', 'hermes-3-flash'], 'default_max_tokens': 8192, 'signup_url': 'https://nousresearch.com', 'supports_health_check': True, 'model_profiles': {'hermes-4': {'supportsReasoning': True, 'supportsThinking': False, 'combinedBudget': False, 'contextWindow': 131072, 'maxOutputTokens': 8192}, 'hermes-4-flash': {'supportsReasoning': False, 'supportsThinking': False, 'combinedBudget': False, 'contextWindow': 131072, 'maxOutputTokens': 8192}, 'hermes-3': {'supportsReasoning': False, 'supportsThinking': False, 'combinedBudget': False, 'contextWindow': 131072, 'maxOutputTokens': 8192}, 'hermes-3-flash': {'supportsReasoning': False, 'supportsThinking': False, 'combinedBudget': False, 'contextWindow': 131072, 'maxOutputTokens': 8192}, '*': {'supportsReasoning': False, 'supportsThinking': False, 'combinedBudget': False, 'contextWindow': 131072, 'maxOutputTokens': 4096}}}

def resolveBaseUrl() -> str:
    return 'https://api.nousresearch.com/v1'

def resolveApiKey(envKey: str | None=None) -> str | None:
    return envKey